libs/vre/resource-editor/resource-editor/src/lib/resource-fetcher.component.ts

```diff
@@ -1,52 +1,65 @@
-import { ChangeDetectorRef, Component, Input, OnChanges, OnDestroy } from '@angular/core';
+import { Component, Input, OnChanges, OnDestroy } from '@angular/core';
 import { ResourceFetcherService } from '@dasch-swiss/vre/resource-editor/representations';
 import { DspResource } from '@dasch-swiss/vre/shared/app-common';
-import { Subscription } from 'rxjs';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-resource-fetcher',
-  template: ' <app-resource *ngIf="resource" [resource]="resource" />',
+  template: `
+    <ng-container *ngIf="!loading">
+      <app-resource *ngIf="resource; else noResourceTpl" [resource]="resource" />
+    </ng-container>
+
+    <ng-template #noResourceTpl
+      ><h3 style="text-align: center; margin-top: 50px">This resource does not exist.</h3>
+    </ng-template>
+  `,
   providers: [ResourceFetcherService],
 })
 export class ResourceFetcherComponent implements OnChanges, OnDestroy {
   @Input({ required: true }) resourceIri!: string;
 
   resource?: DspResource;
+  loading!: boolean;
+  private _ngUnsubscribe = new Subject<void>();
 
-  private _subscription?: Subscription;
-
-  constructor(
-    private _cdr: ChangeDetectorRef,
-    private _resourceFetcherService: ResourceFetcherService
-  ) {}
+  constructor(private _resourceFetcherService: ResourceFetcherService) {}
 
   ngOnChanges() {
+    this.loading = true;
     this._resourceFetcherService.onDestroy();
     this._resourceFetcherService.onInit(this.resourceIri);
 
-    if (this._subscription) {
-      this._subscription.unsubscribe();
-    }
+    this._unsubscribe();
```

Comment:
This _unsubscribe() seems wrong to me. If it works, then only because the resourcefetcher needs more time for the request.

Imho there is also no need to destroy anything or unsubscribe. Subscribing to this._resourceFetcherService.resource$ could happen once. It must happen onInit or in the constructor though.